python/social_bot/envs/grocery_ground.py

```diff
@@ -252,19 +261,40 @@ def task_specific_observation(self):
             np.array of the extra observations should be added into the
             observation besides self states, for the non-image case
         """
-        return self.get_icub_extra_obs(self._agent)
+        icub_extra_obs =  self.get_icub_extra_obs(self._agent)
+        agent_pos = icub_extra_obs[:3]
+        agent_speed = (agent_pos - self._pre_agent_pos) / self._step_time
+        self._pre_agent_pos = agent_pos
+        yaw = self._agent.get_pose()[1][2]
+        target_pos, _ = self._target.get_pose()
+        yaw = yaw % (2 * np.pi) - np.pi  # model icub has a built-in 180 degree rotation
+        walk_target_theta = np.arctan2(target_pos[1] - agent_pos[1], target_pos[0] - agent_pos[0])
```

Comment:
make use of some tricks from [roboschool forward walker](https://github.com/openai/roboschool/blob/42d1c1c14f57b58f156072b6b5f9b761ecaf3018/roboschool/gym_forward_walker.py#L60)